cql3/attributes.cc

```diff
@@ -149,4 +149,26 @@ lw_shared_ptr<column_specification> attributes::raw::timeout_receiver(const sstr
     return make_lw_shared<column_specification>(ks_name, cf_name, ::make_shared<column_identifier>("[timeout]", true), duration_type);
 }
 
+static unsigned make_bit_mask(bool timestamp, bool time_to_live, bool timeout) noexcept {
```

Comment:
And I can't be sure without trying this myself, but looks too me like a very verbose patch just to do run-time validation. I wonder if it's possible to do what I proposed in the beginning - use a defaulted operator== (does this exit?) to check if the attributes are equal to ones with just the allowed attributes.

Reply:
A customized operator== will need to compare the presence of the different attributes rather than their values.
I think it's an overkill.
The runtime check prevents against programming errors, not user errors, with the syntax-level checking.
But since we're only dealing with those 3 kinds of attributes it might be an overkill indeed.

Reply:
bit_masks patch dropped in v4